packages/cli/src/bin/tag-application.ts

```diff
@@ -0,0 +1,55 @@
+import { chalk } from './utils/format.js';
+import { Spinner } from './utils/spinner.js';
+import { resolveAppPackage, resolveAppKey } from '../lib/app-package.js';
+import { appRegistered, validateToken, tagAppBundle, type FusionEnv } from './utils/app-api.js';
+
+enum Tags {
+    preview = 'preview',
+    latest = 'latest',
+}
+
+export const tagApplication = async (options: {
+    tag: keyof typeof Tags;
+    version: string;
+    env: FusionEnv;
+    service: string;
+}) => {
+    const { tag, version, env, service } = options;
+
+    const spinner = Spinner.Global({ prefixText: chalk.dim('Tag') });
+
+    const validToken = validateToken();
+    if (!validToken) {
+        return;
```

Comment:
instead of return, throw error

Reply:
i would like to use exit(1) here. maybe we can adda flag for verbose later on and throw all error then?